packages/vbridge/src/VBridge.ts

```diff
@@ -351,10 +343,50 @@ export class VBridge {
     if (userTokenAllowance.lt(publicAmount)) {
       await tokenInstance.approveSpending(vAnchor.contract.address);
     }
-    // Make Merkle proof
-    const merkleProof = inputs.map((x) => this.getVAnchor(Number(x.originChainId))!.getMerkleProof(x));
-    await vAnchor.bridgedTransact(inputs, outputs, fee, recipient, relayer, merkleProof);
-    await this.updateLinkedVAnchors(vAnchor);
+
+    const regeneratedInputs: Utxo[] = [];
+
+    // Populate the leaves map
+    const leavesMap: Record<string, Uint8Array[]> = {};
+    leavesMap[chainId] = vAnchor.tree.elements().map((commitment) => hexToU8a(commitment.toHexString()))
+
+    for (let input of inputs) {
+      const inputTree = this.getVAnchor(Number(input.originChainId)).tree;
+
+      if (!leavesMap[input.originChainId]) {
+        leavesMap[input.originChainId] = inputTree.elements().map((commitment) => hexToU8a(commitment.toHexString()))
+      }
+
+      // update the utxo with the proper index 
+      const utxoIndex = inputTree.getIndexByElement(u8aToHex(input.commitment));
+      const newUtxo = await CircomUtxo.generateUtxo({
```

Comment:
If we're called this a `CircomUtxo` do you think we need to provide the backend? I would assume its clear that it should automatically be a `'Circom'` backend.